mteb/abstasks/AbsTaskMultilabelClassification.py

```diff
@@ -0,0 +1,146 @@
+from __future__ import annotations
+
+import logging
+from collections import defaultdict
+
+import numpy as np
+from sklearn.preprocessing import MultiLabelBinarizer
+
+from ..evaluation.evaluators import kNNMultiLabelClassificationEvaluator
+from .AbsTask import AbsTask
+
+logger = logging.getLogger(__name__)
+
+
+class AbsTaskMultilabelClassification(AbsTask):
+    """
+    Abstract class for kNN multioutput classification tasks
+    The similarity is computed between pairs and the results are ranked.
+
+    self.load_data() must generate a huggingface dataset with a split matching self.metadata_dict["eval_splits"], and assign it to self.dataset. It must contain the following columns:
+        text: str
+        label: list[Hashable]
+    """
+
+    def __init__(
+        self,
+        n_experiments=None,
+        samples_per_label=None,
+        k=3,
+        batch_size=32,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.batch_size = batch_size
+
+        # Bootstrap parameters
+        self.n_experiments = (
+            n_experiments
+            if n_experiments is not None
+            else self.metadata_dict.get("n_experiments", 10)
+        )
+        self.samples_per_label = (
+            samples_per_label
+            if samples_per_label is not None
+            else self.metadata_dict.get("samples_per_label", 8)
+        )
+        # kNN parameters
+        self.k = k
+        # Run metadata validation by instantiating addressing the attribute
+        # This is quite hacky. Ideally, this would be done in the constructor of
+        # each concrete task, but then we have to duplicate the __init__ method's
+        # interface.
+        if hasattr(self, "metadata"):
+            self.metadata
+
+    def _add_main_score(self, scores):
+        if self.metadata_dict["main_score"] in scores:
```

Comment:
```suggestion
        if self.metadata.main_score in scores:
```
I would use the metadata (no dict) generally. This makes code changes easier in the future.

Reply:
What should I do about these?
```python
        self.n_experiments = (
            n_experiments
            if n_experiments is not None
            else self.metadata_dict.get("n_experiments", 10)
        )
        self.samples_per_label = (
            samples_per_label
            if samples_per_label is not None
            else self.metadata_dict.get("samples_per_label", 8)
        )
```
Would it be smart to write that also over to `metadata`? Like this for instance:
```python
if n_experiments is None:
     try:
          self.n_experiments = self.metadata.n_experiments
     except AttributeError:
          self.n_experiments = 10
else:
     self.n_experiments = n_experiments
```
or perhaps a bit more concise but less readable:
```python
self.n_experiments = n_experiments or getattr(self.metadata, "n_experiments", 10)
```

Reply:
I have removed all references to `metadata_dict` can you check if you like it better now? @KennethEnevoldsen